superset/security/manager.py

```diff
@@ -877,3 +878,34 @@ def assert_viz_permission(self, viz: "BaseViz") -> None:
         """
 
         self.assert_datasource_permission(viz.datasource)
+
+    def get_rls_filters(self, table: "BaseDatasource"):
+        """
+        Retrieves the appropriate row level security filters for the current user and the passed table.
+
+        :param table: The table to check against
+        :returns: A list of filters strings.
+        """
+        if (
+            hasattr(g, "user")
+            and hasattr(g.user, "roles")
+            and hasattr(table, "row_level_security")
```

Comment:
Trying to make sense of all 3 `hasattr` here. So `g.user` should be defined if a user is logged in, and if defined, it seems like `user.roles` should always be defined, unless I'm missing something.

Now I'm guess `row_level_security` has to do with `DruidDatasource`. Should we just add a `row_level_security = []` as a class attribute there (or better, on `BaseDatasource`) for consistency?

Reply:
Certain automated tests failed if i didn't include `hasattr(g.user, "roles")`.  I imagine it was `AttributeError: 'AnonymousUserMixin' object has no attribute 'id'`.

Since they're now retrieved by a query, a lot of this is no longer necessary.